lib/iris/cube.py

```diff
@@ -1632,7 +1638,33 @@ def dtype(self):
 
     @dtype.setter
     def dtype(self, dtype):
-        self._dtype = dtype
+        if dtype != self.dtype:
```

Comment:
Should we not also allow here `cube.dtype = None`, to reset the dtype to the underlying data ?